### PR TITLE
Frozen Nodes fix

### PR DESCRIPTION
**Pull request by ramramps**
## Purpose
This PR fixes the issue with Geometry disappear (Dynamo standalone) when the graph re-executes. 

This PR refers to the below tasks 
http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9274
http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9267

The Fixes for this PR are : 
1. Loading a graph with Frozen nodes preserves the freeze state of downstream nodes
2. Disconnecting a Frozen node, recomputes the freeze state correctly.

 ## Declaration
Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [] The level of testing this PR includes is appropriate
- [] User facing strings, if any, are extracted into `*.resx` files

## Reviewers
 @mjkkirschner

### CODE REVIEW

File: src/DynamoCore/Graph/Nodes/NodeModel.cs [hunk 1]
```diff
@@ -678,20 +678,12 @@ public bool IsFrozen
         /// <returns></returns>
         internal bool IsAnyUpstreamFrozen()
         {
+            var test = this.Name;
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Nodes/NodeModel.cs]: lets get rid of this debugging line

File: src/DynamoCore/Graph/Nodes/NodeModel.cs [hunk 2]
```diff
@@ -866,7 +857,7 @@ public virtual void OnUpdateASTCollection()
         /// silenced. This is particularly critical for code block nodes, whose modification can 
         /// mutate the workspace.
         /// 
-        /// As opposed to RaisesModificationEvents, this modifies the entire parent workspace
+        /// As opposed to RaisesModificationEvents, this modifies the entire parent worksfpace
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Nodes/NodeModel.cs]: type

File: src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs [hunk 3]
```diff
@@ -1543,17 +1544,13 @@ internal IEnumerable<NodeModel> GetSourceNodes()
         private void ComputeUpstreamCacheForEntireGraph()
         {
             //get the source nodes or roots of the DAG
-            var sources = GetSourceNodes();
-            var allVisited = new HashSet<NodeModel>();
-            foreach(var source in sources)
+            //var sources = GetSourceNodes();
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs]: can get rid of commented out code here

File: src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs [hunk 4]
```diff
@@ -1543,17 +1544,13 @@ internal IEnumerable<NodeModel> GetSourceNodes()
         private void ComputeUpstreamCacheForEntireGraph()
         {
             //get the source nodes or roots of the DAG
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs]: can get rid of this comment

File: src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs [hunk 5]
```diff
@@ -1543,17 +1544,13 @@ internal IEnumerable<NodeModel> GetSourceNodes()
...
-            var sources = GetSourceNodes();
-            var allVisited = new HashSet<NodeModel>();
-            foreach(var source in sources)
+            //var sources = GetSourceNodes();
+            var sortedNodes = AstBuilder.TopologicalSort(this.nodes);             
+            
+            foreach (var sortedNode in sortedNodes)
             {
                 //call computeUpstreamOnDownstream to propogate the upstream Cache down to all nodes
```
Comment (mjkkirschner) [inline src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs]: outdated comment

**Comment by mjkkirschner** [comment]
a few small comments

**Comment by ramramps** [comment]
@mjkkirschner - Done

**Pull request merged**